apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/dao/DexOfferTable.java

```diff
@@ -0,0 +1,81 @@
+import com.apollocurrency.aplwallet.apl.core.db.LongKey;
+import com.apollocurrency.aplwallet.apl.core.db.LongKeyFactory;
+import com.apollocurrency.aplwallet.apl.core.db.dao.mapper.DexOfferMapper;
+import com.apollocurrency.aplwallet.apl.exchange.model.DexOffer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Objects;
+
+/**
+ * Implemented for backward compatibility with rollback function in the DerivedDbTable.
+ * Use DexOfferDao
+ */
+@Deprecated
+@Singleton
+public class DexOfferTable  extends EntityDbTable<DexOffer> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DexOfferTable.class);
+
+    static final LongKeyFactory<DexOffer> KEY_FACTORY = new LongKeyFactory<>("transaction_id") {
+        @Override
+        public DbKey newKey(DexOffer offer) {
+            return new LongKey(offer.getTransactionId());
+        }
+
+        @Override
+        public DexOffer newEntity(DbKey dbKey) {
+            DexOffer offer = new DexOffer();
+            offer.setTransactionId(((LongKey) dbKey).getId());
+            return offer;
+        }
+    };
+
+
+    private static final String TABLE_NAME = "dex_offer";
+    private final Blockchain blockchain;
+    private DexOfferMapper dexOfferMapper;
+
+    @Inject
+    public DexOfferTable(Blockchain blockchain, DexOfferMapper dexOfferMapper) {
+        super(TABLE_NAME, KEY_FACTORY, true, "",false);
+        this.dexOfferMapper = dexOfferMapper;
+        this.blockchain = Objects.requireNonNull(blockchain, "Blockchain is NULL");
+    }
+
+    @Override
+    protected DexOffer load(Connection con, ResultSet rs, DbKey dbKey) throws SQLException {
+        return dexOfferMapper.map(rs, null);
+    }
+
+    @Override
+    protected void save(Connection con, DexOffer offer) throws SQLException {
+        try (PreparedStatement pstmt = con.prepareStatement("INSERT INTO dex_offer (transaction_id, account_id, type, " +
+                "offer_currency, offer_amount, pair_currency, pair_rate, finish_time, status, height, latest)" +
+                        "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, TRUE)")){
+            int i = 0;
+            pstmt.setLong(++i, offer.getTransactionId());
+            pstmt.setLong(++i, offer.getAccountId());
+            pstmt.setByte(++i, (byte) offer.getType().ordinal());
+            pstmt.setByte(++i, (byte) offer.getOfferCurrency().ordinal());
+            pstmt.setLong(++i, offer.getOfferAmount());
+            pstmt.setByte(++i, (byte) offer.getPairCurrency().ordinal());
+            pstmt.setLong(++i, offer.getPairRate());
+            pstmt.setInt(++i, offer.getFinishTime());
+            pstmt.setByte(++i, (byte) offer.getStatus().ordinal());
+            pstmt.setInt(++i, blockchain.getHeight());
```

Comment:
Better to use height field from your model instance, instead of calling blockchain.getHeight()

Reply:
I don't think that DexOffer should know block height information.

Reply:
Dex offer represents db model and should know values, which will be saved to db